include/openssl/bio.h

```diff
@@ -724,10 +724,16 @@ void BIO_copy_next_retry(BIO *b);
  * long BIO_ghbn_ctrl(int cmd,int iarg,char *parg);
  */
 
-# ifdef __GNUC__
-#  define __bio_h__attr__ __attribute__
-# else
-#  define __bio_h__attr__(x)
+# define __bio_h__attr__(x)
+# if defined(__GNUC__) && defined(__STDC_VERSION__)
+    /*
+     * Because we support the 'z' modifier, which made its appearance in C99,
+     * we can't use __attribute__ with pre C99 compilers.
```

Comment:
&& isn't the correct linguistic interpretation.  It's a logical implication.  Cause => effect.

Regarding information, I got it from [here](https://www.gnu.org/software/libc/manual/html_node/Integer-Conversions.html):

> ‘z’ was introduced in ISO C99.

But, I'm changing "compiler" to "dialect"